system/jlib/jptree.cpp

```diff
@@ -7630,27 +7733,15 @@ void mergeConfiguration(IPropertyTree & target, IPropertyTree & source)
         //Legacy support for old roxie configuration files that have repeated elements with no name tag
         if (!name)
             name = child.queryProp("@netAddress");
-        const char * path = tag;
-        if (name)
-        {
-            tempPath.clear().append(path).append("[@name=\'").append(name).append("']");
-            path = tempPath;
-        }
-        if (child.queryProp("@__remove__"))
-        {
-            target.removeProp(path);
-        }
-        else
-        {
-            IPropertyTree * match = target.queryPropTree(path);
-            if (!match)
-            {
-                match = target.addPropTree(tag);
-                if (name)
-                    match->setProp("@name", name);
-            }
-            mergeConfiguration(*match, child);
-        }
+
+        bool first = false;
+        bool endprior = false;
+        bool sequence = checkInSequence(child, seqname, first, endprior);
+        if (first && isScalarItem(child))
```

Comment:
Right, this implementation does not remove arrays of structures (objects).  I kept the logic that if it finds a matching named item it will be merged like a singleton object would be.  I can change it to only merge singletons if preferred, or I can only merge root level.